crates/transaction-pool/src/maintain.rs

```diff
@@ -494,6 +495,82 @@ pub async fn maintain_transaction_pool<N, Client, P, St, Tasks>(
 
                 // keep track of mined blob transactions
                 blob_store_tracker.add_new_chain_blocks(&blocks);
+
+                // At Osaka transition, we need to convert blobs to new format
+                if !chain_spec.is_osaka_active_at_timestamp(tip.timestamp()) &&
+                    chain_spec.is_osaka_active_at_timestamp(tip.timestamp().saturating_add(12))
+                {
+                    let pool = pool.clone();
+                    let spawner = task_spawner.clone();
+                    let client = client.clone();
+                    task_spawner.spawn(Box::pin(async move {
+                        let mut interval = tokio::time::interval(Duration::from_secs(1));
+                        loop {
+                            // Loop and replace blob transactions until we reach Osaka transition
+                            // block after which no legacy blobs are going to be accepted.
+                            let last_iteration =
+                                client.latest_header().ok().flatten().is_none_or(|header| {
+                                    client
+                                        .chain_spec()
+                                        .is_osaka_active_at_timestamp(header.timestamp())
+                                });
```

Comment:
this exit condition assumes we're making it past the block, if that doesn't happen this loop will keep going, but since we have the timeout this is fine